qa/smoke-test-http/src/test/java/org/elasticsearch/http/snapshots/RestGetSnapshotsIT.java

```diff
@@ -176,7 +178,24 @@ public void testSortAndPaginateWithInProgress() throws Exception {
             inProgressSnapshots.add(AbstractSnapshotIntegTestCase.startFullSnapshot(logger, repoName, snapshotName, false));
         }
         AbstractSnapshotIntegTestCase.awaitNumberOfSnapshotsInProgress(logger, inProgressCount);
-
+        AbstractSnapshotIntegTestCase.awaitClusterState(logger, state -> {
+            boolean firstIndexSuccessfullySnapshot = state.custom(SnapshotsInProgress.TYPE, SnapshotsInProgress.EMPTY)
+                .asStream()
+                .flatMap(s -> s.shards().stream())
+                .allMatch(
+                    e -> e.getKey().getIndexName().equals("test-index-1") == false
+                        || e.getValue().state() == SnapshotsInProgress.ShardState.SUCCESS
+                );
+            List<SnapshotsInProgress.ShardState> secondIndexStates = state.custom(SnapshotsInProgress.TYPE, SnapshotsInProgress.EMPTY)
+                .asStream()
+                .flatMap(s -> s.shards().stream())
+                .filter(e -> e.getKey().getIndexName().equals("test-index-2"))
+                .map(e -> e.getValue().state())
+                .collect(Collectors.toList());
+            boolean secondIndexIsBlocked = secondIndexStates.stream().filter(e -> e == SnapshotsInProgress.ShardState.INIT).count() == 1
+                && secondIndexStates.stream().filter(e -> e == SnapshotsInProgress.ShardState.QUEUED == false).count() == 1;
```

Comment:
I think we should instead check that the number of queued states is `inProgressCount - 1`? That would be a bit clearer I think.

Perhaps we could use `Collectors.groupingBy` to group by status and check that we have 1 INIT, `inProgressCount - 1` QUEUED and nothing else?